### PR TITLE
Add Unsafe.coerce

### LINKED ISSUE #494: Maybe test failure
I get the following output for the maybe.carp test:

```
Test 'nothing? works on Nothing' passed
Test 'nothing? works on Just' passed
Test 'just? works on Just' passed
Test 'just? works on Nothing' passed
Test 'from works on Just' passed
Test 'from works on Nothing' passed
Test 'unsafe-from works on Just' passed
Test 'apply works on Just' passed
Test 'apply works on Nothing' passed
Test 'or-zero works on Just' passed
Test 'or-zero works on Nothing' passed
Test 'to-result works on Just' passed
Test 'to-result works on Nothing' passed
Test 'ptr works on Just' failed:
	Expected value: '1', actual value: '0'
Test 'ptr works on Nothing' passed
Test 'from-ptr works on Ptr/Just' passed
Test 'from-ptr works on NULL/Nothing' passed
Results:
	|=================|
	Passed: 16	Failed: 1
[RUNTIME ERROR] '"./out/Untitled"' exited with return value 1.
```

### PR DESCRIPTION
**Pull request by jacereda**
Used for Maybe.ptr, fixes #494.

### CODE REVIEW

File: core/Interfaces.carp [hunk 1]
```diff
@@ -4,6 +4,8 @@
 ;;(definterface copy (λ [&a] a))
 ;;(definterface str (λ [a] String))
 
+(definterface coerce (λ [a] b))
```
Comment (hellerve) [inline core/Interfaces.carp]: This is only needed if there are different implementations for this for different types. Is that going to be how this is used? My understanding was that `Unsafe.coerce` kind of stood on its own.

File: core/Maybe.carp [hunk 2]
```diff
@@ -55,9 +55,10 @@ It is the inverse of [`just?`](#just?).")
   (doc ptr "Creates a `(Ptr a)` from a `(Maybe a)`. If the `Maybe` was
 `Nothing`, this function will return a `NULL` value.")
   (defn ptr [a]
-    (match a
-      (Nothing) NULL
-      (Just x) (address x)))
+    (the (Ptr a) (Unsafe.coerce
+      (match @a
```
Comment (eriksvedang) [inline core/Maybe.carp]: There are plans to make matching work on refs very soon, this would use that – right?

File: core/Interfaces.carp (hunk 1 above)
Comment (jacereda) [reply core/Interfaces.carp]: Oh, right. Fixed.

File: core/Maybe.carp (hunk 2 above)
Comment (jacereda) [reply core/Maybe.carp]: Can I read about that somewhere?

Comment (eriksvedang) [reply core/Maybe.carp]: https://github.com/carp-lang/Carp/issues/469

Comment (jacereda) [reply core/Maybe.carp]: Yeah, I guess the implementation would change then.

**Pull request merged**